### PR TITLE
Support customizable Faraday connection options

### PR DESCRIPTION
**Pull request by raedmakhlouf**
**Description**
Allow for more customizable `Faraday::ConnectionOptions` when calling Adyen's API.

At a high level, our set-up (and perhaps that of other future partners of Adyen) leverages a third-party vaulting provider for tokenization. In order to use the vaulting provider's offering, we need to be able to proxy requests through them first, before they forward the request over to Adyen.

In doing so, the Adyen client has been updated here to accept a customizable set of Faraday connection options to pass along to Faraday.

**Tested scenarios**
- Defaults to what was happening before, if no options are provided, with `Faraday::ConnectionOptions.new`.
- The Faraday connection is created with the options that are passed in.

**Fixed issue**:  I have not created a GH issue for this feature request, but can if required.

### CODE REVIEW

**Review by michaelpaul — APPROVED** [review]

**Comment by raedmakhlouf** [comment]
thanks for the review @michaelpaul!

could I get a second review from another code-owner? 🙂

**Review by wboereboom — APPROVED** [review]
Thanks for the contribution. Looks good to me!

**Pull request merged**
